src/formatDate.js

```diff
@@ -8,7 +8,47 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldYearIndex = getIndex(fromFormat);
+  const newYearIndex = getIndex(toFormat);
+  const oldYearLength = fromFormat[oldYearIndex].length;
+  const newYearLength = toFormat[newYearIndex].length;
+  const oldJoin = fromFormat[3];
+  const newJoin = toFormat[3];
+  const newDate = date.split(oldJoin);
+  const reorderedDate = [];
+  const oldYear = newDate[oldYearIndex];
+
+  function getIndex(index) {
```

Comment:
The getIndex function is being declared inside the formatDate function. It's better to declare helper functions outside the main function to improve readability and avoid unnecessary redeclaration every time the main function is called.